source/common/router/router.cc

```diff
@@ -540,12 +555,26 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers,
     }
   }
 
-  Http::ConnectionPool::Instance* http_pool = getHttpConnPool();
+  absl::variant<Http::ConnectionPool::Instance*, Tcp::ConnectionPool::Instance*> conn_pool;
   Upstream::HostDescriptionConstSharedPtr host;
+  bool should_tcp_proxy = shouldTcpProxy(*downstream_headers_, route_entry_);
 
-  if (http_pool) {
-    host = http_pool->host();
+  if (!should_tcp_proxy) {
```

Comment:
Do we need any stats for the CONNECT cases, both termination and pass through? Any other missing stats? Fine to do this stuff in a follow up but might be worth a TODO for this type of stuff. I assume also that all of the circuit breakers are built into the TCP connection pool so we are good there?

Reply:
We've got general upgrade stats, but I didn't split out CONNECT or differentiate connect upgrade from connect termination.  Have thoughts on what folks would use and I can add them in a follow up?

Reply:
Ultimately I think having some stats specific to CONNECT/WS might be useful, but I agree we can see what people want later.